### PR TITLE
Add multi project optimisation option to allow opting into shared tiny remapper.

### PR DESCRIPTION
**Pull request by modmuss50**
Im not 100% sure on the name, happy for other suggestions.

This option should be enabled in projects with many (5+) sub projects. This has been made opt-out due to fundamental issues raised. The peformance hit for most people with no sub projects will be 0, projects with a small handful of subprojects should only be a tiny bit slower.

### CODE REVIEW

**Comment by Juuxel** [comment]
> The peformance hit for most people with no sub projects will be 0

Might actually be faster if there's one less task to run (haven't checked yet)

File: src/main/java/net/fabricmc/loom/task/AbstractRemapJarTask.java [hunk 1]
```diff
@@ -228,4 +229,9 @@ protected static Function<File, String> relativePath(List<String> rootPaths) {
 			return s;
 		};
 	}
+
+	@Internal
+	protected LoomGradleExtension getLoomExtension() {
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/task/AbstractRemapJarTask.java:236]: This should also be `@ApiStatus.Internal`

File: src/main/java/net/fabricmc/loom/LoomGradleExtension.java [hunk 2]
```diff
@@ -108,4 +108,6 @@ default List<Path> getMinecraftJars(MappingsNamespace mappingsNamespace) {
 	boolean refreshDeps();
 
 	void setRefreshDeps(boolean refreshDeps);
+
+	boolean multiProjectOptimisation();
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/LoomGradleExtension.java:118]: Would benefit from a javadoc comment like this:

```
If true, multi-project optimisation mode is enabled. This mode makes builds with many Loom projects
much faster by increasing sharing and disabling some functionality.

<p>You can enable it by setting the Gradle property {@code fabric.loom.multiProjectOptimisation} to {@code true}.
```

**Review by Juuxel — CHANGES_REQUESTED** [review]

**Comment by modmuss50** [comment]
Going to merge, test failures are unrealted. 👍

**Pull request merged**